Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -159,6 +164,45 @@ private MapIcon parsePoint() throws IOException, XmlPullParserException, KMLPars
             + mParser.getPositionDescription());
   }
 
+  @NonNull
+  private MapPolyline parseLineString()
+      throws IOException, XmlPullParserException, KMLParseException {
+    mParser.require(XmlPullParser.START_TAG, mNameSpace, "LineString");
+    while (mParser.next() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      String type = mParser.getName();
+      if (type.equals("coordinates")) {
+        AltitudeReferenceSystemWrapper altitudeReferenceSystemWrapper =
+            new AltitudeReferenceSystemWrapper(AltitudeReferenceSystem.GEOID);
+        ArrayList<Geoposition> positions = parseCoordinates(altitudeReferenceSystemWrapper);
+        if (positions.size() < 2) {
+          throw new KMLParseException(
+              "coordinates for a LineString must contain at least two positions. Instead saw: "
+                  + positions.size()
+                  + " at position: "
+                  + mParser.getPositionDescription());
+        }
+        ParsingHelpers.setAltitudesToZeroIfAtSurface(
+            positions, altitudeReferenceSystemWrapper.getAltitudeReferenceSystem());
+        MapPolyline line = mFactory.createMapPolyline();
+        line.setPath(
+            new Geopath(positions, altitudeReferenceSystemWrapper.getAltitudeReferenceSystem()));
+        mParser.require(XmlPullParser.END_TAG, mNameSpace, "LineString");
```

Comment:
This looks like it will fail if coordinates is not the last tag in LineString.
This will fail:
<LineString>
  <coordinates>stuff goes here</coordinates>
  <altitudeOffset>...</altitudeOffset>
</LineString>

Add a test case for this please.

Reply:
Github at my formatting. Trying HTML
&lt;LineString&gt;
  &lt;coordinates&gt;...&lt;coordinates&gt;
  &lt;otherValidTag /%gt;
&lt;LineString&gt;

Reply:
Tags must appear in a specified order, and the coordinates tag will always be last, directly before </LineString>